conjur/policy/policy_logic.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+"""
+PolicyLogic module
+
+This module is the business logic for executing the POLICY command
+"""
+# Builtins
+import json
+import logging
+
+# pylint: disable=too-few-public-methods
+class PolicyLogic:
+    """
+    PolicyLogic
+
+    This class holds the business logic for executing and manipulating
+    returned data
+    """
+    def __init__(self, client):
+        self.client = client
+
+    def run_action(self, policy_data):
+        """
+        Method to determine which subcommand action to run {apply, replace, update}
+        """
+        # pylint: disable=logging-fstring-interpolation
+        logging.debug(f"Running '{policy_data.action}' for file '{policy_data.file} ' \
+                      f'under '{policy_data.branch}'")
+        if policy_data.action == 'replace':
+            resources = self.client.replace_policy_file(policy_data.branch, policy_data.file)
```

Comment:
also not for this PR but I think http requests should be async. we should make ourselves fermilliar with the async await keywords and the asyncio module.

Reply:
fabulous idea